docs/use-case-descriptions/message-validation.md

```diff
@@ -13,25 +13,73 @@ If it fails validation, a response message is sent with ‘rejected’ in the re
-| Unknown reference <Reference\>  | The message with the sequence where is referred to is unknown.</br>For <Reference\> the concerning reference field name can be filled in (for example FlexRequestSequence or PrognosisSequence).                     |
-| Reference Period mismatch        | The message(s) with the sequence where is referred to contains a different Period.                                                                                                                                    |
-| Reference message expired        | The message that is referred to is expired.                                                                                                                                                                           |
-| Reference message revoked        | The message that is referred to is revoked.                                                                                                                                                                           |
-| ISPs out of bounds               | One or more ISPs are outside the tolerated boundaries: ISPs do not exist.                                                                                                                                             |
-| ISP conflict                     | One or more ISPs are defined more than once, possibly because of an incorrect duration.                                                                                                                               |
-| Period out of bounds             | Period of the message is inappropriate.</br>For example: a FlexRequest with a Period in the past or a settlement item in a FlexSettlement with a Period outside the concerning settlement period.                     |
-| ExpirationDateTime out of bounds | ExpirationDateTime is in the past or exceeds the ISPs in the message.                                                                                                                                                 |
+
+| RejectionReason | Cause of rejection | Applicable to messages |
+|---|---|---|
+| Unknown SenderDomain | SenderDomain specified in the SignedMessage is unknown to the receiving implementation. | SM |
+| Invalid SenderRole | The SenderRole specified in the SignedMessage is invalid. | SM |
+| Invalid Sender combination | The combination SenderDomain and SenderRole in the SignedMessage is invalid.| SM |
+| Mismatch SenderDomain | SenderDomain specified in the SignedMessage doesn't match with the SenderDomain in the inner XML message. | SM & All other message types |
+| Invalid SignedMessage | Despite being schema-compliant, the syntax, type or semantics of the SignedMessage were unacceptable for the receiving implementation. | SM |
+| Invalid Message | Despite being schema-compliant, the syntax, type or semantics of the message were unacceptable for the receiving implementation. | All message types |
```

Comment:
This message is really different from the original message; and now there is no message that states "There is a mismatch between the SenderDomain/Role combination in the message wrapper and the inner XML message. " => the latter seems to be a viable error case that warrants its own error message?

Reply:
"There is a mismatch between the SenderDomain/Role combination in the message wrapper and the inner XML message. "
In the inner message types there is no SenderRole attribute so there can't be a mismatch between the outer SenderDomain/Role combination and the inner one.

The separation of "Invalid SignedMessage" and "Invalid Message" was done so you then specifically know at what point the validation fails. If it's in the wrapper (that might be a generic issue) or within the message itself.

What do you think above the above reasoning?

Reply:
Ok, I agree with your response; nevertheless I have a language issue: the line "_Despite being schema-compliant, the syntax, type or semantics of the SignedMessage were unacceptable for the receiving implementation._" is a bit strange from an English language point of view; can't you make it something like "_The SignedMessage complies with the XSD schema, but failed to pass validation_"

Reply:
Updated to your suggestion, I agree that that's better understandable language. :-)